clang-tools-extra/clangd/unittests/ReplayPeambleTests.cpp

```diff
@@ -42,7 +41,11 @@
 #include <memory>
 #include <vector>
 
-namespace clang::clangd {
+namespace clang {
+
+class Module;
```

Comment:
this is not an "unnecessary" include, clangd requires a declaration of `clang::Module` in this file (even if incomplete) and we try not to have forward declarations.

what's the motivation behind this change exactly?

Reply:
Is that a clangd preference or llvm-wide? I see a lot of forward declarations in other parts of the code so I want to understand when I should and shouldn't use them in this project.

The motivation is that I'm making a bunch of changes to Module.h, and reducing the number of files that include it speeds up my local development (and full builds, too) by requiring less rebuilding.

Reply:
Specifically I'm following the guidance at: https://llvm.org/docs/CodingStandards.html#include-as-little-as-possible

Reply:
yes that's mostly something we try in clangd, but fair point, feel free to ignore this one.

i was mostly pushing because this is just a leaf hence having that dependency won't affect compile times drastically, and we'll probably keep pulling it through other headers anyway. (e.g. even after this change, we still have `clangd/SourceCode.h -> clang/Lex/HeaderSearch.h -> clang/Lex/DirectoryLookup.h -> clang/Lex/ModuleMap.h -> clang/Basic/Module.h, hence this isn't really changing much in practice, while hiding the dependency)